src/odo.ts

```diff
@@ -361,14 +373,25 @@ export class OdoImpl implements Odo {
     public async _getComponents(application: OpenShiftObject): Promise<OpenShiftObject[]> {
         const result: cliInstance.CliExitData = await this.execute(Command.listComponents(application.getParent().getName(), application.getName()));
         let data: any[] = [];
+        let compSource = '';
         try {
             data = JSON.parse(result.stdout).items;
         } catch (ignore) {
             // show no apps if output is not correct json
             // see https://github.com/openshift/odo/issues/1521
         }
-        const apps: string[] = data.map((value) => value.metadata.name);
-        return apps.map<OpenShiftObject>((value) => new OpenShiftObjectImpl(application, value, ContextType.COMPONENT, this, TreeItemCollapsibleState.Collapsed));
+        const apps = data.map(value => ({ name: value.metadata.name, source: value.spec.source }));
+        return apps.map<OpenShiftObject>((value) =>
+            {
+                if (value.source.startsWith('https://')) {
+                    compSource = 'git';
+                } else if (value.source.endsWith('.jar|')) {
```

Comment:
binary components can be created with any file, not only jar ones.

Reply:
yeah I need to add the condition to test all the probabilities. Need to figure out a way to get them using a regex or something.

Reply:
Updated with the fix.